nautobot_ssot/migrations/0007_remove_dashed_custom_fields.py

```diff
@@ -0,0 +1,50 @@
+from django.db import migrations
+from nautobot.extras.utils import FeatureQuery
+
+CF_KEY_CHANGE_MAP = {
+    "servicenow_last_synchronized": "ssot-synced-to-servicenow",
+    "ssot_last_synchronized": "ssot-synced-from-ipfabric",
+    "ipfabric_site_id": "ipfabric-site-id",
+}
+
+
+def migrate_dashed_custom_fields_to_underscored(apps, schema_editor):
+    CustomField = apps.get_model("extras", "customfield")
+    ContentType = apps.get_model("contenttypes", "ContentType")
+
+    for key in CF_KEY_CHANGE_MAP.values():
+        for custom_field in CustomField.objects.filter(key=key):
+            original_key = custom_field.key
+            updated_key = CF_KEY_CHANGE_MAP[key]
+            print(
+                f'CustomField instance "{custom_field.label}" key attribute "{original_key}" is being changed to "{updated_key}".'
+            )
+            custom_field.key = updated_key
+            custom_field.save()
+
+    for ct in ContentType.objects.filter(FeatureQuery("custom_fields").get_query()):
+        relevant_custom_fields = CustomField.objects.filter(content_types=ct)
```

Comment:
Perhaps narrow this down further with something like `.filter(content_types=ct, key__in=CF_KEY_CHANGE_MAP.values())` so that we are just iterating the models that these specific CFs apply to rather than all models that have *any* CF defined?

Reply:
Good idea, done!